code/modules/mob/dead/observer/observer.dm

```diff
@@ -365,6 +365,9 @@ Works together with spawning an observer, noted above.
 
 	ghost.set_huds_from_prefs()
 
+	if(iszombie(src)&&stat == !DEAD)
```

Comment:
Even when not inhabited by a client, a zombie can still be reviving (generally this should only not be occurring if we're decapitated); so you might want to either also check if we're reviving, or have the revival perform a check if we have a client and then also free_for_ghosts there if there is no client.